src/sentry/relay/config/__init__.py

```diff
@@ -417,9 +417,26 @@ def _filter_option_to_config_setting(flt, setting):
 CUSTOM_MEASUREMENT_LIMIT = 5
 
 
+class CustomMeasurementSettings(TypedDict):
+    limit: int
+
+
+class TransactionMetricsSettings(TypedDict):
+    extractMetrics: Collection[str]
+    extractCustomTags: Collection[str]
+    customMeasurements: CustomMeasurementSettings
+
+
 def get_transaction_metrics_settings(
     project: Project, breakdowns_config: Optional[Mapping[str, Any]]
-):
+) -> TransactionMetricsSettings:
+
+    if killswitches.killswitch_matches_context(
+        "relay.drop-transaction-metrics", {"project_id": project.id}
+    ):
+        # Do not extract anything
+        return {"extractMetrics": [], "extractCustomTags": [], "customMeasurements": {"limit": 0}}
```

Comment:
This conditional is very similar to the feature flag check: https://github.com/getsentry/sentry/blob/1eca234240cfbc1b56216dba8bcee720064c59fc/src/sentry/relay/config/__init__.py#L443

But I guess it makes sense to have a kill switch mechanism independent of the feature flag?

Reply:
i think we should unify those checks tbh. i don't think it's a good idea to send more possible values that mean the same thing, because the potential is tehre that this doesnt parse correctly.

Reply:
I was hoping to prevent that by adding type hints to the return value, but I agree that a single check makes more sense, will refactor.